core/store/models/job_spec.go

```diff
@@ -41,7 +41,7 @@ type TaskSpecRequest struct {
 // for a given contract. It contains the Initiators, Tasks (which are the
 // individual steps to be carried out), StartAt, EndAt, and CreatedAt fields.
 type JobSpec struct {
-	ID         string      `json:"id,omitempty" gorm:"primary_key;not null"`
+	ID         string      `json:"id" gorm:"primary_key;not null"`
```

Comment:
Think you removed the wrong `omitempty`. I'll revisit it for you.

Reply:
:flushed: good catch